packages/blade/src/components/Menu/_decisions/decisions.md

```diff
@@ -0,0 +1,308 @@
+          <MenuItem title="Audio" />
+        </MenuOverlay>
+      </Menu>
+    </MenuBody>
+    <MenuFooter>
+      <Button>Apply</Button>
+    </MenuFooter>
+  </MenuOverlay>
+</Menu>
+```
+
+## Props
+
+### Menu
+
+```ts
+type MenuProps = {
+  /**
+   * First children is trigger and second children is MenuOverlay
+   **/
+  children: [React.ReactElement, React.ReactElement];
+
+  /**
+   * Open controlled state
+   */
+  isOpen?: boolean;
+
+  /**
+   * On Menu open change callback
+   */
+  onOpenChange?: (isOpen: boolean) => void;
+
+  /**
+   * Should menu open on click or hover
+   *
+   * @default 'click'
+   */
+  openInteraction?: 'hover' | 'click';
+};
+```
+
+### MenuOverlay
+
+This overlay will be flexible overlay which can have any component. Static / Non-Static / MenuItem / etc
+
+```ts
+type MenuOverlay = {
+  /**
+   * A slot inside Menu's overlay.
+   *
+   * Supports MenuItem or any other component
+   */
+  children: React.ReactElement[];
+};
+```
+
+### MenuItem
+
+```ts
+type MenuItemProps = {
+  /**
+   * title of item
+   */
+  title: string;
+
+  /**
+   * Description text for the item
+   */
+  description?: string;
+
+  /**
+   * Click handler for MenuItem
+   *
+   * Absense of this prop and href will turn the item into non-interactive item
+   */
+  onClick?: (event: React.MouseEvent) => void;
+
+  /**
+   * Link to open when item is clicked.
+   *
+   * Absense of this prop and onClick will turn the item into non-interactive item
+   */
+  href?: string;
+
+  /**
+   * HTML target of the link
+   */
+  target?: string;
+
+  /**
+   * Item that goes on left-side of item.
+   *
+   * Will be overriden in multiselect
+   */
+  leading?: React.ReactNode;
+
+  /**
+   * Item that goes on right-side of item.
+   */
+  trailing?: React.ReactNode;
+
+  /**
+   * Item that goes immediately next to the title.
+   */
+  titleSuffix?: React.ReactElement;
+
+  /**
+   * disabled state of item
+   */
+  isDisabled?: boolean;
+
+  /**
+   * Color of item. set to negative for dangerous actions like Delete, Remove, etc
+   */
+  color?: Extract<FeedbackColors, 'negative'>;
+};
+```
+
+### Menu Subcomponent Props
+
+```ts
+// Follows the standard header, footer types of Blade like DropdownHeader, ModalHeader, etc
+type MenuHeaderProps = BaseHeaderProps;
+type MenuFooterProps = BaseFooterProps;
+
+// No props
+type MenuDividerProps = {};
+```
+
+## Example APIs
+
+### Basic API
+
+```jsx
+<Menu>
+  <Button>Edit</Button> {/* Can be Link, Avatar, or any custom interactive item */}
+  <MenuOverlay>
+    <MenuBody>
+      {/* Supports any JSX */}
+      <Box>Slot</Box>
+      <Divider />
+      <Box overflowY="auto">
+        <MenuItem title="Profile" />
+        <MenuItem>
+          <Text>Custom Slot in Item</Text>
+        </MenuItem>
+      </Box>
+    </MenuBody>
+  </MenuOverlay>
+</Menu>
+```
+
+### SubMenu
+
+```jsx
+<Menu>
+  <Button>Edit</Button>
+  <MenuOverlay>
+    <MenuItem title="Profile" />
```

Comment:
I remember we discussed around having a wrapper to consolidate the MenuItems in a section which would make it easy for you to add negative margins? 

```
<MenuOverlay>
  // any jsx
  <MenuList>
    <Menu />
  </MenuList>
  // any jsx
</MenuOverlay>
```

Reply:
Discussed in office that day. MenuList can't isolate the list navigation in Menu's case and for negative margins, I added MenuDivider component instead so it handles the styling otherwise it was too much of styling just to get end-to-end divider.

Also exposed bleed prop on Divider but we need to know the padding for us to add correct negative margins and position absolute was not working as expected because then it depends on parent position relative